### PR TITLE
Azure Simple Baseline Template

### PR DESCRIPTION
**Pull request by erikschlegel**
## All Submissions:
-------------------------------------
* [YES] Have you followed the guidelines in our Contributing [document](./CONTRIBUTING.md)?
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [YES] I have updated the documentation accordingly.
**Comments** - I updated most of the documentation with the exception of the traffic manager modules. I'll include reference readmes for both the tm-profile and tm-endpoint modules in a follow-up PR. 
* [NO] I have added tests to cover my changes.
**Comments** - We'll issue a follow-up PR for adding automated tests for `azure-simple` once @nmiodice completes #90 
* [NA] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [NA] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

## What is the current behavior?
-------------------------------------
This pull request accounts for supporting a single region azure simple cobalt template. 
* At the moment, this template supports a single rest-based CRUD container deployed to a standard linux app service plan.
* We had to make a number of changes across the below modules to support an end-to-end app service deployment that's integrated with app-gateway and traffic manager as the single-only endpoint for inbound traffic. 

    - app-gateway
    - app-service
    - app-service-plan
    - app-insights

* This pull request closes out the first acceptance criteria from user story #91 `Internet traffic to backend APIs are blocked through the vnet access restriction feature`
* There will be multiple PR's to address the remaining acceptance criteria tasks from story #91

Issue Number: #91 

## What is the new behavior?
-------------------------------------
You can follow the sample usage [here](https://github.com/microsoft/cobalt/tree/erisch/features/azure-simple/infra/templates/azure-simple#example-usage) to deploy this template to a target azure subscription 

## Does this introduce a breaking change?
-------------------------------------
- [NO]

## Any relevant logs, error output, etc?
-------------------------------------
- The `GET` request for the backend API can be tested by referencing the traffic manager dns: http://cobalt-azure-simple-dns.trafficmanager.net 
- Direct access to the backend API is restricted to the vnet gateway: https://cobalt-backend-api.azurewebsites.net 
- Here's a screen shot of the deployed resources 
image

### CODE REVIEW

File: infra/templates/azure-simple/variables.tf [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  default     = ["app-gateway"]
+}
+
+variable "subnet_prefixes" {
+  description = "The address prefix to use for the subnet."
+  default     = ["10.0.1.0/24"]
+}
+
+variable "appgateway_frontend_port_name" {
+  description = "Name that will be given to the front-end port"
```
Comment (TechnicallyWilliams) [inline infra/templates/azure-simple/variables.tf]: Is this variable missing a default value?

File: infra/templates/azure-simple/main.tf [hunk 2]
```diff
@@ -0,0 +1,39 @@
+locals {
+  service_plan_name          = "${var.name}-sp"
```
Comment (TechnicallyWilliams) [inline infra/templates/azure-simple/main.tf]: Is the "service_plan_name" value satisfied via command line argument?

File: infra/templates/azure-simple/variables.tf [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# Resource Group
+variable "resource_group_location" {
+  description = "The deployment location of resource group container all the resources"
+  type        = "string"
+}
+
+variable "name" {
```
Comment (TechnicallyWilliams) [inline infra/templates/azure-simple/variables.tf]: Is the "service_plan_name" value satisfied via command line argument?

File: infra/templates/azure-simple/variables.tf (hunk 1 above)
Comment (erikschlegel) [reply infra/templates/azure-simple/variables.tf]: `appgateway_frontend_port_name` is a required configuration parameter

**Comment by erikschlegel** [comment]
> Got errors deploying plan:
> 
> ```
> Error: Error applying plan:
> 
> 4 error(s) occurred:
> 
> * module.app_service.azurerm_app_service.appsvc: 1 error(s) occurred:
> 
> * azurerm_app_service.appsvc: The name "cobalt-backend-api" used for the App Service needs to be globally unique and isn't available: Hostname 'cobalt-backend-api' already exists. Please select a different name.
> * module.traffic_manager_profile.azurerm_traffic_manager_profile.profile: 1 error(s) occurred:
> 
> * azurerm_traffic_manager_profile.profile: trafficmanager.ProfilesClient#CreateOrUpdate: Failure sending request: StatusCode=409 -- Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
> * module.app_gateway.data.azurerm_subnet.appgateway: data.azurerm_subnet.appgateway: Error: Subnet "app-gateway" (Virtual Network "cobalt-azure-simple-vnet" / Resource Group "cobalt-azure-simple-ip-test-rg") was not found
> * module.traffic_manager_endpoint.azurerm_public_ip.pip: 1 error(s) occurred:
> 
> * azurerm_public_ip.pip: Error Creating/Updating Public IP "cobalt-azure-simple-ip" (Resource Group "cobalt-azure-simple-ip-test-rg"): network.PublicIPAddressesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="DnsRecordInUse" Message="DNS record cobalt-azure-simple-ip-dns.eastus.cloudapp.azure.com is already used by another public IP." Details=[]
> 
> Terraform does not automatically rollback in the face of errors.
> Instead, your Terraform state file has been partially updated with
> any resources that successfully completed. Please address the error
> above and apply again to incrementally change your infrastructure.
> ```

@iphilpot this has been resolved in my most recent commit. You can go ahead and run `destroy` and `apply` to re-confirm.

**Comment by iphilpot** [comment]
Still getting errors... look new:

```
Error: Error applying plan:

3 error(s) occurred:

* module.traffic_manager.azurerm_traffic_manager_profile.profile: 1 error(s) occurred:

* azurerm_traffic_manager_profile.profile: trafficmanager.ProfilesClient#CreateOrUpdate: Failure sending request: StatusCode=409 -- 
Original Error: autorest/azure: Service returned an error. Status=<nil> <nil>
* module.traffic_manager.azurerm_public_ip.pip: 1 error(s) occurred:

* azurerm_public_ip.pip: Error Creating/Updating Public IP "cobalt-azure-simple-ip" (Resource Group "cobalt-azure-simple-rgip"): network.PublicIPAddressesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="DnsRecordInUse" Message="DNS record cobalt-azure-simple-ip-dns.eastus.cloudapp.azure.com is already used by another public IP." Details=[]
* module.app_service.azurerm_app_service.appsvc: 1 error(s) occurred:

* azurerm_app_service.appsvc: The name "cobalt-backend-api" used for the App Service needs to be globally unique and isn't available: Hostname 'cobalt-backend-api' already exists. Please select a different name.
```

**Pull request merged**